### PR TITLE
mqtt: Move conf code to rust

### PR DESCRIPTION
**Pull request by jlucovsky**
Continuation of #10085 

IIssue: 6387

This commit moves the configuration logic to Rust.

Link to [redmine](https://redmine.openinfosecfoundation.org/projects/suricata/issues) ticket: [6387](https://redmine.openinfosecfoundation.org/issues/6387)

Describe changes:
- Move configuration file code to rust
- Make max_msg_size a u32 (was usize)
- Removed extraneous #includes from C module.

Updates:
- Removed MQTT C source modules
- Registration uses Rust registration function.

### Provide values to any of the below to override the defaults.

To use a pull request use a branch name like `pr/N` where `N` is the
pull request number.

Alternatively, `SV_BRANCH` may also be a link to an
OISF/suricata-verify pull-request.

```
SV_REPO=
SV_BRANCH=
SU_REPO=
SU_BRANCH=
LIBHTP_REPO=
LIBHTP_BRANCH=
```

### CODE REVIEW

**Review by jasonish — APPROVED** [review]
Looks good to me.

File: rust/src/mqtt/mqtt.rs [hunk 1]
```diff
@@ -112,7 +112,7 @@ pub struct MQTTState {
     connected: bool,
     skip_request: usize,
     skip_response: usize,
-    max_msg_len: usize,
+    max_msg_len: u32,
```
Comment (catenacyber) [inline rust/src/mqtt/mqtt.rs:115]: Why this change ?

**Comment by catenacyber** [comment]
Looking good, just added one question

**Comment by suricata-qa** [comment]
WARNING:
 
 | field | baseline | test | % |
|--|--|--|--| 
| | SURI_TLPW2_autofp_stats_chk | | |
| .uptime | 181 | 192 | 106.08% | 

Pipeline 17179

File: rust/src/mqtt/mqtt.rs (hunk 1 above)
Comment (jlucovsky) [reply rust/src/mqtt/mqtt.rs:115]: I didn't see a good reason for keeping `max_msg_len` as a `usize` (which seems more for pointers than scalars).

File: rust/src/mqtt/mqtt.rs [hunk 2]
```diff
@@ -830,6 +828,13 @@ pub unsafe extern "C" fn rs_mqtt_register_parser(cfg_max_msg_len: u32) {
                 SCLogError!("Invalid value for mqtt.max-tx");
             }
         }
+        if let Some(val) = conf_get("app-layer.protocols.mqtt.max-msg-length") {
+            if let Ok(v) = get_memval(val) {
+                MAX_MSG_LEN = v as u32;
+            } else {
+                SCLogError!("Invalid value for mqtt.max-msg-length: {}", val);
```
Comment (victorjulien) [inline rust/src/mqtt/mqtt.rs:835]: not new in this PR, but this is an error message acting like a warning. I think we have several more of these, this deserves a ticket I think.

**Review by victorjulien — APPROVED** [review]

**Comment by victorjulien** [comment]
Merged in #10100, thanks!

File: rust/src/mqtt/mqtt.rs (hunk 2 above)
Comment (jlucovsky) [reply rust/src/mqtt/mqtt.rs:835]: [Issue 6652](https://redmine.openinfosecfoundation.org/issues/6652)
